### PR TITLE
serverless: read a configuration file while running as a Lambda Extension.

### PR DESCRIPTION
**Pull request by remeh**
### What does this PR do?

The Datadog Lambda Extension will try to read the configuration
file in `/var/task/datadog.yaml`.

### Motivation

It will help us use some nice features such as a the logs scrubbing.

### Additional Notes

I've documented its usage in the matching `README.md`.
I've also removed the expvar HTTP server which is not used by the Serverless Agent.

### Describe your test plan

N/A

### CODE REVIEW

File: cmd/serverless/main.go [hunk 1]
```diff
@@ -231,10 +221,11 @@ func runAgent(ctx context.Context, stopCh chan struct{}) (err error) {
 	// read configuration from the environment vars
 	// --------------------------------------------
 
-	// note that this call is counter-intuitive: it must return an error because
-	// no files should exist, and then, the configuration is read from env vars.
-	if _, confErr := config.Load(); confErr == nil {
-		log.Warn("A configuration file has been found, which should not happen in this mode.")
+	// we will look for the configuration file in the /var/task folder,
+	// where AWS Lambda should make it available.
+	config.Datadog.SetConfigFile("/var/task/datadog.yaml")
```
Comment (DarcyRaynerDD) [inline cmd/serverless/main.go:226]: We could reuse the environment variable: 'datadogConfigPath'. Also, looking at the code I believe there may be a bug with how the trace agent config is loaded. The value of 'datadogConfigPath' should be "/var/task/datadog.yaml"

**Review by DarcyRaynerDD — CHANGES_REQUESTED** [review]

File: cmd/serverless/main.go (hunk 1 above)
Comment (remeh) [reply cmd/serverless/main.go]: I've changed the value of the const and I'm now using it! This should address both concerns.

**Comment by nhinsch** [comment]
Looking at the [example configuration file](https://github.com/DataDog/datadog-agent/blob/master/pkg/config/config_template.yaml), I see that there are a *lot* of possible options. Presumably not all of these apply in the Serverless context. Should we document which options are supported? What is the worst that can happen if an unsupported option is added?

File: cmd/serverless/main.go [hunk 2]
```diff
@@ -231,10 +224,9 @@ func runAgent(ctx context.Context, stopCh chan struct{}) (err error) {
 	// read configuration from the environment vars
```
Comment (nhinsch) [inline cmd/serverless/main.go:224]: I would update this comment

**Review by DarcyRaynerDD — APPROVED** [review]

File: cmd/serverless/main.go (hunk 2 above)
Comment (remeh) [reply cmd/serverless/main.go:224]: Good catch. Updating that.

**Comment by remeh** [comment]
@nhinsch that's a very good question: before sending that PR I took a quick look at that file as well. I think that for most of these parameters, not much unintended would happen or at least nothing surprising, ofc if some of them are wrongly used the extension will misbehave and a lot of them makes no sense in the context of the extension.

My opinion is that we should document only the ones that make sense but I don't feel like we need to implement a whitelist, but it's really not a strong opinion. I like the idea that we may be able to leverage some of these existing fields in the future without having to release a new version.

File: cmd/serverless/README.md [hunk 3]
```diff
@@ -13,7 +13,52 @@ GOOS=linux go build -tags serverless -o datadog-agent
...
+Another example, if you want to exclude some logs line containing a pattern:
+
+```
+logs_config:
+    processing_rules:
+    - type: exclude_at_match
+      name: exclude_datadoghq_users
+      # regexp can be anything
+      pattern: \w+@datadoghq.com
+```
```
Comment (hush-hush) [inline cmd/serverless/README.md:54]: Do we want duplicate example from the doc ? In case of a documentation update this might not be edited. I think the idea is to have one place of truth with the documentation website.

File: cmd/serverless/main.go [hunk 4]
```diff
@@ -85,7 +84,10 @@ where they can be graphed on dashboards. The Datadog Serverless Agent implements
 
 	logsLogsTypeSubscribed = "DD_LOGS_CONFIG_LAMBDA_LOGS_TYPE"
 
-	datadogConfigPath        = "datadog.yaml"
+	// AWS Lambda is writing the Lambda function files in /var/task, we want the
```
Comment (hush-hush) [inline cmd/serverless/main.go:87]: Is this definitive from AWS ? Could we let user modify this through an env var (without documenting it for now). This could be useful if one day AWS chose to change this to avoid a force upgrade of the agent.

File: cmd/serverless/main.go [hunk 5]
```diff
@@ -228,13 +221,13 @@ func runAgent(ctx context.Context, stopCh chan struct{}) (err error) {
...
+	// read configuration from both the environment vars and the config file
+	// if one is provided
+	// --------------------------
 
-	// note that this call is counter-intuitive: it must return an error because
-	// no files should exist, and then, the configuration is read from env vars.
-	if _, confErr := config.Load(); confErr == nil {
-		log.Warn("A configuration file has been found, which should not happen in this mode.")
+	config.Datadog.SetConfigFile(datadogConfigPath)
+	if _, confErr := config.LoadWithoutSecret(); confErr == nil {
```
Comment (hush-hush) [inline cmd/serverless/main.go:229]: Could people add their secret executable to the lambda root and be able to use the secret function ? This might be for another PR.

**Review by hush-hush — APPROVED** [review]
Added a few questions, looks good otherwise.

File: cmd/serverless/main.go (hunk 4 above)
Comment (remeh) [reply cmd/serverless/main.go:87]: I think it's fairly definitive indeed since it is how it works and how it is documented on AWS Lambda environments, it was already the case even before the appearance of the extension stuff. We would probably be noticed in time if such a big change would occur (+ releasing a new layer for that is shorter than a full agent release for now).

File: cmd/serverless/README.md (hunk 3 above)
Comment (remeh) [reply cmd/serverless/README.md:54]: I also had mixed feeling concerning shipping samples here copied from the doc. I'll remove them.

File: cmd/serverless/main.go (hunk 5 above)
Comment (remeh) [reply cmd/serverless/main.go:229]: Discussed offline but for now, we won't use the `secrets` feature in the Lambda Extension. It should not be a big deal to enable it but it would be for another PR 👍
